web/services/management/commands/removeoldreports.py

```diff
@@ -0,0 +1,86 @@
+from django.core.management.base import BaseCommand
+from services.models import ErrorReport, UserDetails
+from datetime import timedelta
+from django.utils import timezone
+
+DEFAULT_RECOVERY_FILE_AGE_DAYS = 90
+
+
+class Command(BaseCommand):
+    help = 'Removes identifiable information from ErrorReports (over 90 days \
+            old by default), and removes UserDetails \
+            not related to any active reports. '
+
+    def add_arguments(self, parser):
+        parser.add_argument('--all',
+                            action='store_true',
+                            dest='all',
+                            help='remove all reports and user details')
+
+        parser.add_argument('days', type=int, nargs='?',
+                            help='Number of days to save',
+                            default=DEFAULT_RECOVERY_FILE_AGE_DAYS)
+
+    def handle(self, *args, **options):
+
+        """Get All or Old reports and delete them"""
+
+        all_reports = ErrorReport.objects.all()
+
+        if options['all']:
+            # Choose all reports
+            reports = all_reports
+
+        else:
+            # Choose reports older than a number of days, 90 by default
+            reports = ErrorReport.objects.filter(
+                dateTime__lte=timezone.now()-timedelta(days=options['days']))
+
+        # Delete identifiable parts of chosen reports
+        reports.update(uid="")
+        reports.update(host="")
+        reports.update(user_id="")
+        reports.update(textBox="")
+        reports.update(stacktrace="")
+
+        """Get UserDetails which are not tied to any surviving ErrorReports"""
+
+        users_with_surviving_reports = ErrorReport.objects.exclude(
```

Comment:
... call the `UserDetailsModel` to `removeOrphaned()` with something similar to [this SO answer](https://stackoverflow.com/a/6205303)

This gets SQL to do all the heavy lifting, which it was designed for ;)
```python
from django.db.models import Count

class UserDetails(models.Model):
    def clearOrphanedRecords():
        # First sort all UserDetailRecords by their F.K. count "user`, then using a named pipe
        # Filter out all records equal to 0
        no_refs = UserDetails.objects.annotate(num_refs=Count('user')).filter(num_refs__eq=0)
        no_refs.delete()
```

Reply:
I like this change! I have had to use `no_refs = UserDetails.objects.annotate(num_refs=Count('errorreports')).filter(num_refs=0)`
('user' to 'errorreports' and no `__eq`) for it to work!

Reply:
Ahh I had the join backwards - thanks for working through my slightly mislead guidance

Reply:
Not saying my code doesn't need a once over (a bit freaked out looking at python after 3 weeks of cpp learning), but your comments were enough for me to figure it out!